tests/unit_tests/test_aggregate_temporal_period.py

```diff
@@ -0,0 +1,603 @@
+import json
+
+import pytest
+
+from tests.utils import load_process_code, load_datacube_code, run_process, run_input_validation
+
+
+@pytest.fixture
+def aggregate_temporal_period_process_code():
+    return load_process_code("aggregate_temporal_period")
+
+
+@pytest.mark.parametrize(
+    "data,period,reducer,dimension,context,additional_code_specific_to_test_case,expected_output",
+    [
+        (
+            [{"B01": 1, "B02": 2}],
+            "day",
+            "({data, context}) => data.reduce((acc, val, i, arr) => (acc + val / arr.length) * -1, 0)",
```

Comment:
Why not `-1`? IMO its just a reducer function, as long as it somehow handles the data, it does not really matter if it is `mean` or `sum`  or anything else as long as it gets the job done. If you believe it will be more comprehendable/logical, I will change it.